integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientExternalIdTest.java

```diff
@@ -0,0 +1,320 @@
+        assertNotNull(clientUpdateResponse.getResourceExternalId());
+        assertEquals(clientExternalId, clientUpdateResponse.getResourceExternalId());
+
+        GlobalConfigurationHelper.manageConfigurations(requestSpec, responseSpec,
+                GlobalConfigurationHelper.ENABLE_AUTOGENERATED_EXTERNAL_ID, false);
+    }
+
+    @Test
+    public void testDeleteClientUsingExternalId() {
+        // given
+        ClientHelper clientHelper = new ClientHelper(requestSpec, responseSpec);
+        GlobalConfigurationHelper.manageConfigurations(requestSpec, responseSpec,
+                GlobalConfigurationHelper.ENABLE_AUTOGENERATED_EXTERNAL_ID, true);
+        String jsonPayload = ClientHelper.getBasicClientAsJSON(ClientHelper.DEFAULT_OFFICE_ID, ClientHelper.LEGALFORM_ID_PERSON, null);
+        // when
+        final PostClientsResponse clientResponse = ClientHelper.addClientAsPerson(requestSpec, responseSpec, jsonPayload);
+        final String clientExternalId = clientResponse.getResourceExternalId();
+        jsonPayload = clientHelper.getCloseClientAsJSON();
+        PostClientsClientIdResponse commandResponse = ClientHelper.performClientActionUsingExternalId(requestSpec, responseSpec,
+                clientExternalId, ClientHelper.CLOSE_CLIENT_COMMAND, jsonPayload);
+        jsonPayload = clientHelper.getReactivateClientAsJSON();
+        commandResponse = ClientHelper.performClientActionUsingExternalId(requestSpec, responseSpec, clientExternalId,
+                ClientHelper.REACTIVATE_CLIENT_COMMAND, jsonPayload);
+
+        // then
+        final DeleteClientsClientIdResponse clientDeleteResponse = ClientHelper.deleteClient(requestSpec, responseSpec, clientExternalId);
+        assertNotNull(clientDeleteResponse);
+        assertNotNull(clientDeleteResponse.getResourceExternalId());
+        assertEquals(clientExternalId, clientDeleteResponse.getResourceExternalId());
+
+        GlobalConfigurationHelper.manageConfigurations(requestSpec, responseSpec,
+                GlobalConfigurationHelper.ENABLE_AUTOGENERATED_EXTERNAL_ID, false);
+    }
+
+    @Test
+    public void testGetClientAccountsUsingExternalId() {
+        // given
+        GlobalConfigurationHelper.manageConfigurations(requestSpec, responseSpec,
+                GlobalConfigurationHelper.ENABLE_AUTOGENERATED_EXTERNAL_ID, true);
+        final String jsonPayload = ClientHelper.getBasicClientAsJSON(ClientHelper.DEFAULT_OFFICE_ID, ClientHelper.LEGALFORM_ID_PERSON,
+                null);
+        // when
+        final PostClientsResponse clientResponse = ClientHelper.addClientAsPerson(requestSpec, responseSpec, jsonPayload);
+        final String clientExternalId = clientResponse.getResourceExternalId();
+
+        GetClientsClientIdAccountsResponse clientAccountsResponse = ClientHelper.getClientAccounts(requestSpec, responseSpec,
+                clientExternalId);
+
+        // then
+        assertNotNull(clientAccountsResponse);
```

Comment:
Please check whether it retrieves the proper client (id and externalId)

Reply:
The accounts does not returns any client data, It just returns arrays of loan accounts or savings accounts, etc